apps/src/templates/sectionProgressV2/OpenAllStudentProgressButton.jsx

```diff
@@ -0,0 +1,149 @@
+import throttle from 'lodash/debounce';
+import PropTypes from 'prop-types';
+import React, {useState, useEffect, useRef} from 'react';
+import {connect} from 'react-redux';
+
+import PopUpMenu from '@cdo/apps/lib/ui/PopUpMenu';
+import {studentShape} from '@cdo/apps/templates/teacherDashboard/teacherSectionsRedux';
+import i18n from '@cdo/locale';
+
+import FontAwesome from '../FontAwesome';
+import {
+  collapseMetadataForStudents,
+  expandMetadataForStudents,
+} from '../sectionProgress/sectionProgressRedux';
+
+import style from './expand-all-rows-dropdown.module.scss';
+
+const DROPDOWN_OFFSET = 172;
+
+function OpenAllStudentProgressButton({
+  students,
+  expandMetadataForStudents,
+  collapseMetadataForStudents,
+}) {
+  const [opened, setOpened] = useState(false);
+  const [menuLocation, setMenuLocation] = useState({menuTop: 0, menuLeft: 0});
+  const elementRef = useRef(null);
+  const resizeListener = useRef(null);
+
+  const getAllStudentIds = students.map(student => student.id);
```

Comment:
```suggestion
  const allStudentIds = React.useMemo(() => students.map(student => student.id));
```

If we useMemo than we only have to compute this list once and it will be cached.